Programs/brlapi_client.c

```diff
@@ -1662,10 +1662,29 @@ static int getControllingTty(void)
     FILE *stream;
 
     if ((stream = fopen("/proc/self/stat", "r"))) {
-      int vt;
-      int ok = fscanf(stream, "%*d %*s %*c %*d %*d %*d %d", &tty) == 1;
+      char buf[256];
+      size_t read = fscanf(buf, sizeof(*buf), sizeof(buf) / sizeof(buf[0]), stream);
       fclose(stream);
 
+      /* The line format is "$pid ($comm) $state ...
+       * where comm may contain space as part of its name, ex:
+       * 12345 (someproc) S 123 456) R ...
+       * So to process valid identified after comm, we should skip comm by
+       * getting the 1st ) from end.
```

Comment:
I'd rather parse for ')' starting from the beginning, since we don't know what could be introduced at the end of this file

Reply:
`)` could also be a valid character of comm (example in comment). If this was parsed with 1st ')' from begining, that would lead to state being S and pid being 123 and so on... which would also be incorrect.

The reason for choosing ')' from end is the fact that whatever follows comm in an integer value or whitespace (https://man7.org/linux/man-pages/man5/proc.5.html, from: `/proc/[pid]/stat`).

Reply:
> `)` could also be a valid character of comm (example in comment). 

Uh... (now I have read the debian bug...)

> whatever follows comm is an integer value or whitespace

That's what it is now only, things can and have been added there. I see people saying on the Internet that adding a string with `)` would make it ambiguous and thus that can't happen. Well, the kernel was supposed to respect the documentation (can be parsed with %s) but they didn't.

I guess we indeed have no choice but parse `)` from the right indeed.

Reply:
Then also update the comment: not only `where comm may contain space as part of its name`, but also mention right parentheses.

Reply:
Totally agree !!